src/components/rocp_sdk/rocp_sdk.c

```diff
@@ -477,128 +463,33 @@ check_n_initialize(void)
 int
 check_for_available_devices(char *err_msg)
 {
-    if( PAPI_OK != load_hsa_sym(err_msg) ){
-        return PAPI_EMISC;
-    }
+    int ret_val;
+    struct stat stat_info;
+    const char *dirPath="/sys/class/kfd/kfd/topology/nodes";
 
-    int dev_count = 0;
-    hsa_status_t status = get_device_count(&dev_count);
-    if( status != HSA_STATUS_SUCCESS || dev_count == 0 ) {
-        sprintf(err_msg, "No compatible devices found.");
-        return PAPI_EMISC;
+    // If the path does not exist, there are no AMD devices on this system.
+    ret_val = stat(dirPath, &stat_info);
+    if (ret_val != 0 || !S_ISDIR(stat_info.st_mode)) {
+        goto fn_fail;
     }
 
-    if( unload_hsa_sym() )
-        return PAPI_EMISC;
-
-    return PAPI_OK;
-}
-
-static inline int
-hsa_is_enabled( void )
-{
-    return ( (hsa_initPtr != NULL) &&
-             (hsa_shut_downPtr != NULL) &&
-             (hsa_iterate_agentsPtr != NULL) &&
-             (hsa_agent_get_infoPtr != NULL) );
-}
-
-void *dlopen_from_paths( char *libname, int pathCount, const char *paths[] )
-{
-    void *so = NULL;
-    for (int i = 0; i < pathCount; i++) {
-        if ( paths[i] == NULL ) {
-            continue;
-        }
-        char newpath[PATH_MAX];
-        int count = snprintf(newpath, PATH_MAX, "%s/lib/%s", paths[i], libname);
-        if (count >= PATH_MAX) {
-            SUBDBG("Status string truncated.");
-        }
-        so = dlopen(newpath, RTLD_NOW | RTLD_GLOBAL);
-        if (so) {
-            return so;
-        }
+    // If we can't open this directory, there are no AMD devices on this system.
+    DIR *dir = opendir(dirPath);
+    if (dir == NULL) {
+        goto fn_fail;
     }
-    so = dlopen(libname, RTLD_NOW | RTLD_GLOBAL);
-    return so;
-}
 
-static int
-load_hsa_sym( char *err_msg )
-{
-    const char *paths[] = {getenv("PAPI_ROCP_SDK_ROOT"), getenv("PAPI_ROCM_ROOT")};
-    rocm_dlp = dlopen_from_paths("libhsa-runtime64.so", 2, paths);
-    if (rocm_dlp == NULL) {
-        int count = snprintf(err_msg, PAPI_MAX_STR_LEN, "%s", dlerror());
-        if (count >= PAPI_MAX_STR_LEN) {
-            SUBDBG("Status string truncated.");
-        }
-        return PAPI_EMISC;
+    // If there are no entries in this directory, there are no AMD devices on this system.
+    if( NULL == readdir(dir) ) {
```

Comment:
`readdir` will acknowledge `.` and `..` in a directory. So in the case of `/sys/class/kfd/kfd/topology/nodes`, even if `0`, `1`, `2`, etc are not present `readdir` will not return `NULL` as `.` and `..` are present.